### PR TITLE
Added Link Exit Types

### PR DESCRIPTION
**Pull request by zsweigart**
Adds types used in exit callbacks

### CODE REVIEW

File: types/LinkExitTypes.d.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import { LinkInstitution } from "./LinkInstitution";
+
+export interface LinkExit {
+    error?: LinkError;
+    metadata: LinkExitMetadata;
+}
+
+export interface LinkExitMetadata {
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:8]: order should be 
status
institution
sessionId
requestId
metadataJson

File: types/LinkExitTypes.d.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    link_session_id: string;
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:9]: Is our convention `snake_case` or `camelCase`?
This file contains a mix, whereas https://github.com/plaid/react-native-plaid-link-sdk/pull/234/files was all `camelCase`

Comment (JeroenMols) [reply types/LinkExitTypes.d.ts:9]: sessionId should also be optional

File: types/LinkExitTypes.d.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    request_id: string;
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:10]: Request id should be optional

File: types/LinkExitTypes.d.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    REQUIRES_EXTERNAL_ACTION = 'requires_external_action',
+    REQUIRES_OAUTH = 'requires_oauth',
+    REQUIRES_QUESTIONS = 'requires_questions',
+    REQUIRES_RECAPTCHA = 'requires_recaptcha',
+    REQUIRES_SELECTIONS = 'requires_selections',
+    REQUIRES_DEPOSIT_SWITCH_ALLOCATION_CONFIGURATION = 'requires_deposit_switch_allocation_configuration',
+    REQUIRES_DEPOSIT_SWITCH_ALLOCATION_SELECTION = 'requires_deposit_switch_allocation_selection',
+}
+
+export interface LinkError {
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:30]: Convert to `CamelCase`?

File: types/LinkExitTypes.d.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    error_code: LinkErrorCode;
+    error_type: LinkErrorType;
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:32]: iOS and Android don't have an errorType (For them it's a subtype of ErrorCode)

File: types/LinkExitTypes.d.ts [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    RECAPTCHA_REQUIRED = "RECAPTCHA_REQUIRED",
+    RECAPTCHA_BAD = "RECAPTCHA_BAD",
+
+    // OAUTH_ERROR
+    INCORRECT_OAUTH_NONCE = "INCORRECT_OAUTH_NONCE",
+    OAUTH_STATE_ID_ALREADY_PROCESSED = "OAUTH_STATE_ID_ALREADY_PROCESSED",
+}
+
+export enum LinkErrorType {
+    BANK_TRANSFER = 'BANK_TRANSFER_ERROR',
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:144]: `Bank_Transfer` doesn't exist on Android

File: types/LinkExitTypes.d.ts [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    INVALID_RESULT = 'INVALID_RESULT',
+    INVALID_INPUT = 'INVALID_INPUT',
+    INSTITUTION_ERROR = 'INSTITUTION_ERROR',
+    RATE_LIMIT_EXCEEDED = 'RATE_LIMIT_EXCEEDED',
+    API_ERROR = 'API_ERROR',
+    ITEM_ERROR = 'ITEM_ERROR',
+    AUTH_ERROR = 'AUTH_ERROR',
+    ASSET_REPORT_ERROR = 'ASSET_REPORT_ERROR',
+    SANDBOX_ERROR = 'SANDBOX_ERROR',
+    RECAPTCHA_ERROR = 'RECAPTCHA_ERROR',
```
Comment (JeroenMols) [inline types/LinkExitTypes.d.ts:155]: These also don't exist on Android

File: types/LinkExitTypes.d.ts (hunk 2 above)
Comment (zsweigart) [reply types/LinkExitTypes.d.ts:9]: yea, I'll move these to camelCase

File: types/LinkExitTypes.d.ts (hunk 5 above)
Comment (zsweigart) [reply types/LinkExitTypes.d.ts:32]: I talked with swb and nesting types doesn't make as much sense in ts so we decided to break it back out.  This means the bridge code for Android and iOS will have to break them apart

File: types/LinkExitTypes.d.ts (hunk 6 above)
Comment (zsweigart) [reply types/LinkExitTypes.d.ts:144]: Yep, I found these in the backend code.  We may need to follow up with changes on Android.

File: types/Types.d.ts [hunk 8]
```diff
@@ -127,4 +127,191 @@ export enum LinkAccountSubtype {
     UGMA = 'ugma',
     UTMA = 'utma',
     VARIABLE_ANNUITY = 'variable annuity',
-}
\ No newline at end of file
+}
```
Comment (amytang0) [inline types/Types.d.ts:130]: nit: add line

**Review by amytang0 — APPROVED** [review]
LGTM

**Pull request merged**